### PR TITLE
Edit QA with comments

### PR DESCRIPTION
**Pull request by rpiaggio**
https://github.com/gemini-hlsw/observe/assets/1895643/9d68a42a-5506-4041-b912-2b8344a78cdd

### CODE REVIEW

File: modules/web/client/src/clue/scala/observe/queries/VisitQueriesGQL.scala [hunk 1]
```diff
@@ -31,10 +32,10 @@ object VisitQueriesGQL:
     // ${ExecutionVisitsSubquery.Fragments}
 
   @GraphQL
-  trait UpdateDatasetQAState extends GraphQLOperation[ObservationDB]:
+  trait UpdateDatasetQa extends GraphQLOperation[ObservationDB]:
```
Comment (cquiroz) [inline modules/web/client/src/clue/scala/observe/queries/VisitQueriesGQL.scala:35]: QA?

File: modules/web/client/src/main/scala/observe/ui/components/sequence/steps/QaEditor.scala [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+              InputText(
+                id = "qaComment",
+                value = qaFields.get.comment.map(_.value).orEmpty,
+                placeholder = "Comment",
+                onChange = e =>
+                  qaFields.mod:
+                    EditableQaFields.comment.replace(NonEmptyString.from(e.target.value).toOption)
+              ).withMods(
+                ^.autoComplete.off,
+                ^.onClick ==> ((e: ReactEvent) => e.stopPropagationCB)
```
Comment (cquiroz) [inline modules/web/client/src/main/scala/observe/ui/components/sequence/steps/QaEditor.scala:83]: does `_.stopProp..` work?

**Review by cquiroz — APPROVED** [review]
LGTM

File: modules/web/client/src/main/scala/observe/ui/components/sequence/steps/QaEditor.scala [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+          OverlayPanel(onHide = reset)(
+            <.form(ObserveStyles.QaEditorPanel)(
+              ^.onSubmit --> submit,
+              ^.onKeyDown ==> { e =>
+                e.keyCode match // Allows submitting by pressing enter after changing the status.
+                  case KeyCode.Enter => submit >> panelRef.hide
+                  case _             => Callback.empty
+              }
+            )(
+              SelectButtonOptional[DatasetQaState](
```
Comment (toddburnside) [inline modules/web/client/src/main/scala/observe/ui/components/sequence/steps/QaEditor.scala:60]: Maybe we should add a SelectButtonOptionalEnumView to lucuma-ui sometime.

**Review by toddburnside — APPROVED** [review]
LGTM

File: modules/web/client/src/clue/scala/observe/queries/VisitQueriesGQL.scala (hunk 1 above)
Comment (rpiaggio) [reply modules/web/client/src/clue/scala/observe/queries/VisitQueriesGQL.scala:35]: Not sure how else to group `qaState` + `comment`, so I called it `Qa`.

File: modules/web/client/src/main/scala/observe/ui/components/sequence/steps/QaEditor.scala (hunk 2 above)
Comment (rpiaggio) [reply modules/web/client/src/main/scala/observe/ui/components/sequence/steps/QaEditor.scala]: Actually, it does work but you have to wrap in parens; ie: `^.onClick ==> (_.stopPropagationCB)`. I've pushed a commit.

**Pull request merged**